nomad/structs/services.go

```diff
@@ -283,6 +300,14 @@ func (sc *ServiceCheck) validate() error {
 		return fmt.Errorf("failures_before_critical not supported for check of type %q", sc.Type)
 	}
 
+	// CheckRestart IgnoreWarnings must be true if a check has defined OnUpdate
```

Comment:
I think I understand why this is the case: because if we say the deployment is healthy because we ignored warnings, we would still restart the task and then the deployment status would be incorrect?

This behavior should definitely be documented as well, and that it only flows in one direction. (That is, that you can have a `check_restart` ignore warnings with the default `on_update` behavior.)

Reply:
added a caveat in the docs